lib/pure/options.nim

```diff
@@ -39,6 +39,29 @@ runnableExamples:
 ## inherits from `system.Defect` and should therefore never be caught.
 ## Instead, rely on checking if the option contains a value with the
 ## `isSome <#isSome,Option[T]>`_ and `isNone <#isNone,Option[T]>`_ procs.
+##
+##
+## Pattern matching
+## ================
+##
+## **Note:** This requires the [fusion](https://github.com/nim-lang/fusion) package.
+##
+## [fusion/matching](https://nim-lang.github.io/fusion/src/fusion/matching.html)
+## supports pattern matching on `Option`s, with the `Some(<pattern>)` and
+## `None()` patterns.
+##
+## .. code-block:: nim
+##   {.experimental: "caseStmtMacros".}
+##
+##   import fusion/matching
+##
+##   case some(42)
+##   of Some(@a):
+##     assert a == 42
```

Comment:
```suggestion
##     doAssert a == 42
```
Same the other.

Reply:
but see https://github.com/nim-lang/Nim/pull/17077 ; we should probably update that section in contributing.rst to also cover code-block sections as it should act as `runnableExamples`

Reply:
=> PR https://github.com/nim-lang/Nim/pull/17122